app/questionnaire/questionnaire_schema.py

```diff
@@ -1033,15 +1034,36 @@ def get_answer_format_for_calculated_summary(
         """
         # Type ignore: the block will exist for any valid calculated summary id
         calculated_summary_block: ImmutableDict = self.get_block(calculated_summary_block_id)  # type: ignore
-        first_answer_id = get_calculated_summary_answer_ids(calculated_summary_block)[0]
+        answer_ids = get_calculated_summary_answer_ids(calculated_summary_block)
+        decimal_limit = self.get_decimal_limit(answer_ids)
+        first_answer_id = answer_ids[0]
         first_answer = self.get_answers_by_answer_id(first_answer_id)[0]
         return {
             "type": first_answer["type"].lower(),
             "unit": first_answer.get("unit"),
             "unit_length": first_answer.get("unit_length"),
             "currency": first_answer.get("currency"),
+            "decimal_places": decimal_limit,
         }
 
+    def get_grand_calculated_summary_decimal_limit(
+        self, calculated_summary_ids: list[str]
+    ) -> int:
+        decimal_limits = []
+        # Type ignore: the block will exist for any valid calculated summary id
+        for calculated_summary_id in calculated_summary_ids:
+            answer_ids = get_calculated_summary_answer_ids(self.get_block(calculated_summary_id))  # type: ignore
+            decimal_limits.append(self.get_decimal_limit(answer_ids))
+        return max(decimal_limits)
+
+    def get_decimal_limit(self, answer_ids: list[str]) -> int:
+        decimal_limits: list[int] = [
+            answer.get("decimal_places", 0)
+            for answer_id in answer_ids
+            for answer in self.get_answers_by_answer_id(answer_id)
+        ]
+        return max(decimal_limits)
```

Comment:
We should return None if not set for any. If none of them set it, this would return 0 but we should return None. Will have a chain reaction of some callers.